### PR TITLE
fix: radio instead of dropdown

### PR DESCRIPTION
**Pull request by knhash**
**Feature name:**

    Replace the drop down for password/passphrase option with a radio button choice

**Feature Description**

There are only two options in the drop down, would it be better to replace it with a radio button for choice?
- Aids in discoverability: I completely missed the option for a whole year before I thought to click on the option and see what it contains.
- Less number of interactions: It would take 1 less click to get the same result

Community Forum post: https://community.bitwarden.com/t/radio-choice-password-passphrase/30636

### CODE REVIEW

File: src/popup/generator/password-generator.component.html [hunk 1]
```diff
@@ -21,8 +21,8 @@
         <div class="box-content single-line">
             <a class="box-content-row text-primary" href="#" appStopClick appBlurClick
                 (click)="regenerate()">{{'regeneratePassword' | i18n}}</a>
-            <a class="box-content-row text-primary" href="#" appStopClick appBlurClick
-                (click)="copy()">{{'copyPassword' | i18n}}</a>
+            <a class="box-content-row text-primary" href="#" appStopClick appBlurClick (click)="copy()">{{'copyPassword'
```
Comment (eliykat) [inline src/popup/generator/password-generator.component.html:24]: Something weird has happened here, maybe caused by your Visual Studio Code formatting rules? Please undo this change.

File: src/popup/generator/password-generator.component.html [hunk 2]
```diff
@@ -122,4 +126,4 @@
             </div>
         </div>
     </ng-container>
-</content>
+</content>
```
Comment (eliykat) [inline src/popup/generator/password-generator.component.html:129]: Please undo this change

File: src/popup/generator/password-generator.component.html [hunk 3]
```diff
@@ -38,12 +38,16 @@
             {{'options' | i18n}}
         </div>
         <div class="box-content">
-            <div class="box-content-row" appBoxRow>
-                <label for="type" class="sr-only">{{'type' | i18n}}</label>
-                <select id="type" [(ngModel)]="options.type" (change)="saveOptions()">
-                    <option value="password">{{'password' | i18n}}</option>
-                    <option value="passphrase">{{'passphrase' | i18n}}</option>
-                </select>
+            <div class="box-content-row">
```
Comment (eliykat) [inline src/popup/generator/password-generator.component.html:41]: Please copy across any applicable feedback from here: https://github.com/bitwarden/desktop/pull/955

File: src/popup/generator/password-generator.component.ts [hunk 4]
```diff
@@ -27,6 +28,10 @@ export class PasswordGeneratorComponent extends BasePasswordGeneratorComponent {
 
     async ngOnInit() {
         await super.ngOnInit();
+        this.passTypeOptions = [
```
Comment (eliykat) [inline src/popup/generator/password-generator.component.ts:31]: You can revert the changes in this file once these changes have been lifted up to `jslib`.

**Review by eliykat — CHANGES_REQUESTED** [review]

File: src/popup/generator/password-generator.component.html (hunk 3 above)
Comment (knhash) [reply src/popup/generator/password-generator.component.html:41]: Done. Removed unnecessary for label.

File: src/popup/generator/password-generator.component.html (hunk 1 above)
Comment (knhash) [reply src/popup/generator/password-generator.component.html]: Undone.

File: src/popup/generator/password-generator.component.ts (hunk 4 above)
Comment (knhash) [reply src/popup/generator/password-generator.component.ts]: Done.

File: src/popup/generator/password-generator.component.html [hunk 5]
```diff
@@ -38,12 +38,15 @@
...
+                <div class="radio-group text-default" appBoxRow name="PassTypeOptions"
+                    *ngFor="let o of passTypeOptions">
+                    <input type="radio" [(ngModel)]="options.type" name="Type_{{o.value}}" id="type_{{o.value}}"
+                        [value]="o.value" (change)="saveOptions()" [checked]="options.type === o.value">
+                    <label for="type_{{o.value}}">
+                        {{o.name | i18n}}
```
Comment (eliykat) [inline src/popup/generator/password-generator.component.html:47]: Remove `i18n` pipe per my comment on `desktop`

File: src/popup/generator/password-generator.component.html [hunk 6]
```diff
@@ -38,12 +38,15 @@
             {{'options' | i18n}}
         </div>
         <div class="box-content">
-            <div class="box-content-row" appBoxRow>
-                <label for="type" class="sr-only">{{'type' | i18n}}</label>
```
Comment (eliykat) [inline src/popup/generator/password-generator.component.html:42]: Put this `<label>` back in with some changes - see my comment on `desktop`

**Review by eliykat — CHANGES_REQUESTED** [review]
Same changes as desktop

File: src/popup/generator/password-generator.component.html (hunk 5 above)
Comment (knhash) [reply src/popup/generator/password-generator.component.html]: Done.

File: src/popup/generator/password-generator.component.html (hunk 6 above)
Comment (knhash) [reply src/popup/generator/password-generator.component.html:42]: Done.

**Review by eliykat — APPROVED** [review]

**Pull request merged**